cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -240,6 +240,9 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 		if err != nil {
 			klog.Errorf("AutoscalingStatusProcessor error: %v.", err)
 		}
+
+		// Clear the scale-up failures info each loop so they don't accumulate.
+		a.clusterStateRegistry.ClearScaleUpFailures()
```

Comment:
It seems too explicit.

I suggest you add `clusterStateRegistry.OnLoopStart()`. Then make `ClearScaleUpFailures` private and call it from `OnLoopStart`. That way we will be able to add more "per-loop-cleanup" routines into `OnLoopStart` without overgrowing `clusterStateRegistry` interface

Reply:
Agreed, done!